### PR TITLE
Chesire: Handle creating a new team project

### PR DESCRIPTION
**Pull request by hashtagchris**
`createConnection.js --routing HostIdMapping --project <projectName> ...` shouldn't wait for the service to create the first pipeline. Instead I opted to call createConnection once to create the team project, and again to get the redirect url for creating a pipeline.

### CODE REVIEW

**Comment by hashtagchris** [comment]
How does AEX get the redirect url in this scenario? Or does AEX always call CreateConnection with a projectId for Chesire?

**Comment by jpricketMSFT** [comment]
Yes, in the AEX-Cheshire flow, AEX creates the team project first and then calls CreateConnection.

File: createConnection.js [hunk 1]
```diff
@@ -124,17 +130,26 @@ async function createConnection(inputs) {
...
+                        } else {
+                            console.log(`   Something happened trying to get the definition from url: ${inputs.url}/${inputs.project}/_apis/build/definitions/${definitionId}?propertyFilters=*`);
+                        }
                     }
                 }
+                else {
+                    // "Chesire" flow. Redirect the user to create the first build definition.
+                    console.log('Creating the connection a second time, just to get the redirect url for creating a new build definition.');
+                    inputs.projectId = createdProjectId;
+                    await createConnection(inputs);
```
Comment (jpricketMSFT) [inline createConnection.js]: Seems like this will fail because the Mapping already exists in MPS.

Comment (hashtagchris) [reply createConnection.js]: I think MPS ignores redundant calls, which is good in our case should a customer re-attempt Chesire on error (without bothering to uninstall and reinstall).

**Pull request merged**